experiments/tuning_utils.py

```diff
@@ -554,33 +570,64 @@ def turnovers(results: BacktestResult, prices: pd.DataFrame, test_len: int) -> t
     trades = results.quantities.diff()
     valuation_trades = (trades * prices).dropna()
     relative_trades = valuation_trades.div(results.portfolio_value, axis=0)
-    relative_trades_train = relative_trades.iloc[:-test_len]
-    relative_trades_test = relative_trades.iloc[-test_len:]
 
-    turnover_train = (relative_trades_train.abs().sum(axis=1).mean() * results.periods_per_year) / 2
-    turnover_test = (relative_trades_test.abs().sum(axis=1).mean() * results.periods_per_year) / 2
+    if test_len == 0:
+        relative_trades_train = relative_trades
+        turnover_train = (
+            relative_trades_train.abs().sum(axis=1).mean() * results.periods_per_year
+        ) / 2
+        return turnover_train, None
+
+    elif test_len > 0:
+        relative_trades_train = relative_trades.iloc[:-test_len]
+        relative_trades_test = relative_trades.iloc[-test_len:]
+
+        turnover_train = (
+            relative_trades_train.abs().sum(axis=1).mean() * results.periods_per_year
+        ) / 2
+        turnover_test = (
+            relative_trades_test.abs().sum(axis=1).mean() * results.periods_per_year
+        ) / 2
 
-    return turnover_train, turnover_test
+        return turnover_train, turnover_test
+
+    raise ValueError("test_len must be 0 or positive")
```

Comment:
I think this is slightly easier to read:

```suggestion
    else
        raise ValueError("test_len must be 0 or positive")
```